src/TraceEvent/TraceLog.cs

```diff
@@ -57,21 +57,31 @@ public sealed class TraceLog : TraceEventSource, IDisposable, IFastSerializable,
         /// <returns>The name of the ETLX file that was generated.</returns>
         /// </summary>
         public static string CreateFromEventTraceLogFile(string filePath, string etlxFilePath = null, TraceLogOptions options = null, TraceEventDispatcherOptions traceEventDispatcherOptions = null)
+        {
+            using (TraceEventDispatcher source = TraceEventDispatcher.GetDispatcherFromFileName(filePath, traceEventDispatcherOptions))
```

Comment:
To preserve existing behavior, this method should make sure that `etlxFilePath != null`, and if it is, set it to `etlxFilePath = Path.ChangeExtension(filePath, ".etlx");`.

Reply:
Good catch